### PR TITLE
Implemented attributes

### PR DESCRIPTION
**Pull request by PEMapModder**
- [x] basic attributes system
- [x] hunger mechanism
- [ ] experience mechanism
- [x] effects
  - [x] night vision
  - [x] blindness
  - [x] hunger
    - [x] declaration
    - [x] implementation
- [x] Fix speed-related effects
- [ ] exhaustion
  - [x] attack
  - [x] damage
  - [x] break blocks
  - [ ] walk per meter: 0.01
  - [ ] sneak per meter: 0.005
  - [ ] swim per meter: 0.015
  - [ ] sprint per meter: 0.1
  - [ ] jump: 0.2
  - [x] regen per halfheart | food >= 18: 4.0
- [x] events
  - [x] exhaust event
  - [x] eat event
- [x] data saving

Resolves:
* #3922
* #3830 
* #3809
* #3412
* Possible:
  * #3894

### CODE REVIEW

**Comment by legoboy0215** [comment]
Please add speed attributes

**Comment by legoboy0215** [comment]
Yesss... So only walk exhaustion left?

**Comment by legoboy0215** [comment]
Walking exhaustion needs to be added :P

File: src/pocketmine/item/RawChicken.php [hunk 1]
```diff
@@ -21,10 +21,25 @@
...
+	public function getSaturationRestore() : float{
+		return 1.2;
+	}
+
+	public function getAdditionalEffects() : array{
+		if(mt_rand(0, 9) < 3){
+			return Effect::getEffect(Effect::HUNGER)->setDuration(600);
+		}
+	}
 }
```
Comment (thebigsmileXD) [inline src/pocketmine/item/RawChicken.php]: Somehow forgot RawPorkchop, dude

**Pull request merged**